src/main/kotlin/com/dobby/backend/infrastructure/logging/SensitiveDataMasker.kt

```
@@ -0,0 +1,40 @@
+
+        return try {
+            val jsonString = objectMapper.writeValueAsString(data)
+            var maskedString = jsonString
+            for ((key, maskFunction) in maskPatterns) {
+                maskedString = maskedString.replace(Regex("(?<=\"$key\":\\s?\")[^\"]+")) {
+                    maskFunction(it.value)
+                }
+            }
+
+            objectMapper.readValue<Map<String, Any>>(maskedString)
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)